VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseUnitPagerAdapter.java

```diff
@@ -48,30 +48,41 @@ public CourseComponent getUnit(int pos) {
         return unitList.get(pos);
     }
 
+    /**
+     * @return True if given unit is a video unit (not an only on YouTube unit)
+     */
+    public static boolean isCourseUnitVideo(CourseComponent unit) {
+        return (unit instanceof VideoBlockModel && ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null);
+    }
+
     @Override
     public Fragment getItem(int pos) {
         CourseComponent unit = getUnit(pos);
         CourseUnitFragment unitFragment;
         //FIXME - for the video, let's ignore studentViewMultiDevice for now
-        if (unit instanceof VideoBlockModel &&
-                ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
-            unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
-        } else if (unit instanceof VideoBlockModel &&
-                ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
+        if (isCourseUnitVideo(unit)) {
+            unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit, (pos < unitList.size()), (pos > 0));
+        }
+        else if (unit instanceof VideoBlockModel && ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
```

Comment:
Got some extraneous new lines here, not sure what happend. Could you do an auto-format please?